improver/synthetic_data/set_up_test_cubes.py

```diff
@@ -206,6 +212,84 @@ def construct_scalar_time_coords(time, time_bounds, frt):
     return coord_dims
 
 
+def _create_dimension_coord(
+    coord_array, data, i, coord_name, coord_units, attributes=None
+):
+    """
+    Creates dimension coordinate from coord_array if not None, otherwise creating an array of integers with an interval of 1
+    """
+    if coord_array is not None:
+        if len(coord_array) != data.shape[i]:
+            raise ValueError(
+                "Cannot generate {} {}s from data of shape "
+                "{}".format(len(coord_array), coord_name, data.shape)
+            )
+        coord_array = np.array(coord_array)
+        if issubclass(coord_array.dtype.type, np.integer):
+            # expect integer coord_array
+            coord_array = coord_array.astype(np.int32)
+        else:
+            # option needed for realizations percentile & probability cube setup and heights coordinate
+            coord_array = coord_array.astype(np.float32)
+    else:
+        coord_array = np.arange(data.shape[i]).astype(np.int32)
+    if coord_name in iris.std_names.STD_NAMES:
+        dim_coord = DimCoord(
+            coord_array, coord_name, units=coord_units, attributes=attributes
+        )
+    else:
+        dim_coord = DimCoord(
+            coord_array, long_name=coord_name, units=coord_units, attributes=attributes
+        )
+
+    return dim_coord
+
+
+def _get_dimension_coords(
+    data, y_coord, x_coord, realizations, height_levels, pressure
+):
+    """ Create array of all dimension coordinates. These dimensions will be ordered: realization, height/pressure, y, x. """
+    ndims = len(data.shape)
+    dim_coords = []
```

Comment:
Pedantic but I'd define this after the initial error checking - no point in initialising if an error happens before it's used.

Reply:
Updated [here](https://github.com/metoppv/improver/pull/1309/commits/1d7f21003ca70fbe79b8c8943e280bc37f411b98#diff-ea38258437543d699495d248fce61e29L264).